File: 344. Reverse String/reverse-string.js
// 344. Reverse String (10/1/2021)
// Runtime: 127 ms (46.95%) Memory: 46.52 MB (94.62%) 

/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
function reverseString(s) {
    for (let i = 0; i < parseInt(s.length / 2); i++) [s[i], s[s.length - 1 - i]] = [s[s.length - 1 - i], s[i]];
}

File: 344. Reverse String/reverse-string.c
// 344. Reverse String (10/26/54217)
// Runtime: 48 ms (88.76%) Memory: 12.32 MB (38.80%) 

void reverseString (char* s, int size) {
    int half = size / 2;
    int last = size - 1;
    
    for (int i = 0; i < half; i++) {
        char tmp = s[i];
        s[i] = s[last - i];
        s[last - i] = tmp;
    }
}